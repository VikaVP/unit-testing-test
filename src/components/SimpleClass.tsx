import React from 'react';

export default class SimpleClass extends React.Component<any, any> {
  constructor(props: any) {
    super(props);
    this.state = {
        val: '',
        updateText: ''
    }
    this.handleClick = this.handleClick.bind(this)
    this.handleChange = this.handleChange.bind(this)
  }

  handleClick(){
    this.setState({
        updateText: `Anda menginput ${this.state.val}`
    })
  }

  handleChange(e: any){
    this.setState({
        val: e.target.value
    })
  }

  render() {
    return (
        <>
            <div style={{margin: '50px'}}>
                <input type="text" value={this.state.val} onChange={this.handleChange}/>
            </div>
            <div>
                <button onClick={this.handleClick}>Tes Class</button>
            </div>

            <p style={{margin: '50px'}}>{this.state.updateText}</p>
        </>
    )
  }

}