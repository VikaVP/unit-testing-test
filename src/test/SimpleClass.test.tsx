import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import SimpleClass from '../components/SimpleClass';

//  jest.useFakeTimers();

describe('SimpleClass Page', () => {
  let component: any;
  beforeEach(async () => {
    jest.useFakeTimers();
    const props = {}
    component = shallow(<SimpleClass {...props}/>);
  });

  it('snapshot showing screen', () => {
   expect(shallowToJson(component)).toMatchSnapshot();
   expect(component).toHaveLength(1);
  });

//   it('onchange input', () => {
//     // component.find('input').at(0).prop('onChange')({
//     //     target : {
//     //         value: 'tes class'
//     //     }
//     // })
//     component.find('input').at(0).simulate('change', {
//         target : {
//             value: 'tes class'
//         }
//     })
//     expect(component.find('input').at(0).prop('value')).toBe('tes class')
//   })

  it('onclick button', () => {
    component.find('input').at(0).prop('onChange')({
        target : {
            value: 'tes'
        }
    })
    component.find('button').at(0).prop('onClick')()
    expect(component.find('p').at(0).text()).toEqual('Anda menginput tes')
  })

  it('handleChange spy', () => {
    jest.spyOn(component.instance(), 'handleChange');
    component.instance().handleChange({
        target: {
            value: 'yoo'
        }
    })
  })

})