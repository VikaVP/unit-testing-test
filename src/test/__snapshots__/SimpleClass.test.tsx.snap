// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleClass Page snapshot showing screen 1`] = `
<Fragment>
  <div
    style={
      {
        "margin": "50px",
      }
    }
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Tes Class
    </button>
  </div>
  <p
    style={
      {
        "margin": "50px",
      }
    }
  />
</Fragment>
`;
